### PR TITLE
Convert to Polymer 3, add events type definitions

### LINKED ISSUE #2074: Polymer 3 conversion and TS event type definitions
### Description

- convert codebase to Polymer 3
- use `registerStyles` API instead of `<dom-module theme-for>`
- add events type definitions

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #2074

### CODE REVIEW

File: src/vaadin-grid-scroller.js [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+      }
+    });
+
+    if (this.items && size < this.items.length) {
+      // Size was reduced, scroll to 0 first
+      this._scrollTop = 0;
+    }
+    if (!Array.isArray(this.items)) {
+      // Edge/IE seems to have the lowest maximum
+      const maxVirtualItems = this._edge ? 30000 : 100000;
```
Comment (tomivirkki) [inline src/vaadin-grid-scroller.js:116]: This limit no longer applies for Chromium Edge. 100000 can be used on all supported browsers.

File: src/vaadin-grid.js [hunk 2]
```diff
@@ -0,0 +1,930 @@
...
+      /** @private */
+      _ios: {
+        type: Boolean,
+        value:
+          (/iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream) ||
+          (navigator.platform === 'MacIntel' && navigator.maxTouchPoints > 1)
+      },
+
+      /** @private */
+      _edge: {
```
Comment (tomivirkki) [inline src/vaadin-grid.js:355]: This detects Edge legacy and can be removed.

Comment (web-padawan) [reply src/vaadin-grid.js]: Done.

File: src/vaadin-grid-scroller.js (hunk 1 above)
Comment (web-padawan) [reply src/vaadin-grid-scroller.js]: Done.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**